models/org.go

```diff
@@ -366,6 +367,39 @@ func getOwnedOrgsByUserID(sess *xorm.Session, userID int64) ([]*User, error) {
 		Find(&orgs)
 }
 
+// HasOrgVisible tell if the given user can see one or more of the given orgs
+func HasOrgVisible(orgs []*User, user *User) bool {
+	if len(orgs) == 0 {
+		return false
+	}
+
+	// Not SignedUser
+	if user == nil {
+		for _, org := range orgs {
+			if org.Visibility == VisibleTypePublic {
+				return true
+			}
+		}
+		return false
+	}
+
+	if user.IsAdmin {
+		return true
+	}
+	for _, org := range orgs {
+		switch org.Visibility {
+		case VisibleTypePrivate:
+			if org.IsUserPartOfOrg(user.ID) {
+				return true
+			}
+		default:
+			return true
```

Comment:
Won't this return true no matter if the user is part of the org or not? (Because the default case is `true`)

Reply:
Not necessarily. Because if org is private then first case will match, but in the first case if user is not part of org switch case will complete and below (Line 400) will return false.

Reply:
Then why do we need this switch case if we could also do this (could we?) with something like 
```go
if org.Visibility == VisibleTypePrivate && org.IsUserPartOfOrg(user.ID) {
	return true
}

Reply:
As we need to handle the other types of visibility, so the complete conditional would be:

```go
if (org.Visibility == VisibleTypePrivate && org.IsUserPartOfOrg(user.ID)) || org.Visibility == VisibleTypeLimited || org.Visibility == VisibleTypePublic) {
	return true
}
```
the above switch case is cleaner

Reply:
But where are we explicitly checking these cases? They're not in the switch case, so if `org.Visibility` is either `VisibleTypeLimited` or `VisibleTypePublic` it will return true. IMHO if we're not exiplicitly checking them, it wouldn't make a difference if we included them into a shorter `if`...

Sorry if I'm picking about obvious things here, but I think I don't quite understand your thought.

Reply:
We are checking for both of those visibility types in the switch, they are caught by `default`.